packages/cli/src/services/auth/login.ts

```diff
@@ -0,0 +1,175 @@
+import { homeDir } from '../../utils'
+import {
+  CoginitoAccess,
+  CoginitoSession,
+  connectSessionVerifyDecodeIdToken,
+  DecodedToken,
+  LoginIdentity,
+  ReapitConnectSession,
+} from '@reapit/connect-session'
+import http from 'http'
+import open from 'open'
+import { writeFileSync, promises, existsSync } from 'fs'
+import 'isomorphic-fetch'
+import decode from 'jwt-decode'
+import ora from 'ora'
+
+export class LoginService {
+  private server
+  private readonly connectOAuthUrl: string = 'https://connect.dev.paas.reapit.cloud' // TODO Change dev to env
+  private readonly port: number = 8080
+  private readonly connectUrl: string = `${this.connectOAuthUrl}/login`
+  private readonly redirect = `http://localhost:${this.port}`
+  protected connectSession?: ReapitConnectSession
+
+  static storageLocation: string = `${homeDir()}/.reapit-connect.json`
+
+  private oauthUrl = (clientId: string, authCode: string) =>
+    `${this.connectOAuthUrl}/token?grant_type=authorization_code&client_id=${clientId}&code=${authCode}&redirect_uri=${this.redirect}`
```

Comment:
how do we make sure that the redirect uri is whitelisted in RC? should we have a range that it can pick from (whichever's free) and then whitelist those?